### DISCUSSION TITLE
M2Crypto status

### DISCUSSION THREAD

**Discussion by fstagni** [DIRACGrid/DIRAC#7732] [General]
Hi,
On https://gitlab.com/m2crypto/m2crypto/ README one can read:

> NOTE: This library is currently in maintenance mode. We recommend using a more modern alternative such as [PyCA/cryptography](https://cryptography.io/en/latest/). Examples of how to migrate can be found in [the documentation](https://m2crypto.readthedocs.io/en/latest/howto.migration.html).

The source moved to https://sr.ht/~mcepl/m2crypto/ and all the issues were closed or migrated, e.g. https://todo.sr.ht/~mcepl/m2crypto/278. They provide doc on migration to PyCA: https://m2crypto.readthedocs.io/en/latest/howto.migration.html

We will of course not need this in DiracX, and we have no plans on changing anything in DIRAC in this respect, but DIRAC will anyway need to be supported for a while so a decision will need to be taken. I can't depict all the risks, but at minimum I can think of the case that a new release of OpenSSL breaks something. I see options:
1. Do nothing, and hope nothing happens
2. Fork m2crypto ourselves. Probably still do nothing, but we can have this under control in case something happens.
3. Embark in the migration to PyCA

What do you think is best?

**Comment by chaen** [comment]
Do 1
If things go wrong, do 2
Never do 3

Work on DiracX
